regen-network/regen-web
# perf: use react query in bridge tabs

## Description

Closes: regen-network/regen-registry#1464
- refactor [useBridged](https://github.com/regen-network/regen-web/pull/1771/files#diff-667ba8b62441329d374810f296bb703ff02490065652ceacd4141f436b5fb744) and [getBridgedEcocreditsForAccount](https://github.com/regen-network/regen-web/pull/1771/files#diff-04bbe8e9d39037506257b45de9ca801a42f31ca8d868e0421b78dc0a2dd2b7e2L216-L282) to [useFetchBridgedEcocredits](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468b)
- replace [useBridegeable](https://github.com/regen-network/regen-web/pull/1771/files#diff-3dc5a828ac99557647daabfcd567826ba5aa8a34b25b402052ed5efd1483baec) with [useFetchEcocredits](https://github.com/regen-network/regen-web/pull/1771/files#diff-3c6eb0830ad0060800e8963a3b13df56d6910bbe8569388bde67ad0006b613cb)
- [paginate](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468bR95) BridgeTxStatus queries
- add [polling](https://github.com/regen-network/regen-web/pull/1771/files#diff-e1affe9d17f850340d0b66b414f0fa7717967212aea590c6ba9648a90969468bR97-R104) to pending status along with a [loader and a tooltip](https://github.com/regen-network/regen-web/pull/1771/files#diff-f1f21d0fe8e9869fd0076ecdcd2af7eedafeb9a1a1465c958c53ba27c5885988R88-R104)


---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. Go to https://deploy-preview-1771--regen-registry.netlify.app/ecocredits/bridge
2. Use the shared test account `regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46`
3. Bridge a small amount (ie: `0.001`)
4. Check the `Bridged Credits` tab right after
5. The new tx should appear in pending status and refresh once every 10 seconds

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
